### PR TITLE
Hercules: UT generation implement

### PR DESCRIPTION
**Pull request by DexterDreeeam**
image
image

## Description

Python + LangChain UT generation implementation.
Azure open-ai is considerred a default LLM service.
Only Java source UT generation is supported now.

## Pull Request Checklist

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Code compiles correctly with all tests are passed.
- [ ] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [ ] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [ ] Yes
- [ ] No

## How you tested it
*Please make sure the change is tested, you can test it by adding UTs, do local test and share the screenshots, etc.*

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Technical design
- [ ] Build related changes
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe): 

### Feature UI screenshots or Technical design diagrams
*If this is a relatively large or complex change, kick it off by drawing the tech design with PlantUML and explaining why you chose the solution you did and what alternatives you considered, etc...*

### CODE REVIEW

File: utg2/utg.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    "C:/r/mmx/android/apps/YPC/app/src/main/java/com/microsoft/appmanager",
+]
+src_path = "123C:/r/mmx/android/apps/YPC/app/src/main/java/com/microsoft/appmanager/ViewHelper.java"
+ut_build_s = {
+    "cmd": [r"C:\r\mmx\android\apps\YPC\gradlew.bat", ":app:compileTeamDebugUnitTestSources"],
+    "cwd": r"C:\r\mmx\android\apps\YPC",
+}
+
+repo_paths = [
+    "C:/r/mmx/android/apps/YPC/app/src/main/java/com/microsoft/appmanager",
```
Comment (hydraxman) [inline utg2/utg.py:28]: This is a potential leak of our internal project code structure, please remove them, and you may leverage .gitignore plus a config file to storage these values.

File: utg2/utg.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+from util import *
+from workflow.java.client import client as java_client
+
+repo_paths_s = [
+    "C:/r/mmx/android/apps/YPC/app/src/main/java/com/microsoft/appmanager",
+]
+src_path = "123C:/r/mmx/android/apps/YPC/app/src/main/java/com/microsoft/appmanager/ViewHelper.java"
+ut_build_s = {
+    "cmd": [r"C:\r\mmx\android\apps\YPC\gradlew.bat", ":app:compileTeamDebugUnitTestSources"],
```
Comment (hydraxman) [inline utg2/utg.py:23]: There are many absolute paths, please use var to store the project root and concat it with rel path to get full path.

**Comment by hydraxman** [comment]
rename the folder from utg to hercules

File: utg2/utg.py (hunk 1 above)
Comment (DexterDreeeam) [reply utg2/utg.py]: Fixed

File: utg2/utg.py (hunk 2 above)
Comment (DexterDreeeam) [reply utg2/utg.py]: Fixed

File: .gitignore [hunk 3]
```diff
@@ -72,3 +72,5 @@ env.properties
 
 !/common/src/test/resources/**
 /common/src/test/resources/.DS_Store
+
+/hercules/config.json
```
Comment (hydraxman) [inline .gitignore:76]: If we ignore it, we shouldn't check it in then.

Comment (DexterDreeeam) [reply .gitignore:76]: Fixed, rename config.json to config_template.json

**Review by zhou9584 — APPROVED** [review]

**Pull request merged**